tensorflow/core/transforms/remapper/pass.cc

```diff
@@ -91,18 +96,227 @@ class MatchMulSigmoid : public RewritePattern {
   OperationName sigmoid_name_;
 };
 
+namespace {
+class RewriterBase : public RewritePattern {
+ public:
+  RewriterBase(StringRef opName, OpPropertyHelper &helper,
+               PatternBenefit benefit = PatternBenefit(1))
+      : RewritePattern(opName, benefit, helper.getDialect()->getContext()),
+        helper_(helper),
+        dialect_(helper.getDialect()) {}
+  RewriterBase(MatchAnyOpTypeTag tag, OpPropertyHelper &helper,
+               PatternBenefit benefit = PatternBenefit(1))
+      : RewritePattern(tag, benefit, helper.getDialect()->getContext()),
+        helper_(helper),
+        dialect_(helper.getDialect()) {}
+
+ protected:
+  OpPropertyHelper &helper_;
+  TFGraphDialect *dialect_;
+};
+}  // namespace
+
+static FailureOr<TFOp> CreateContractionBiasAddOp(PatternRewriter &rewriter,
+                                                  OpPropertyHelper &helper,
+                                                  Operation *contraction_op,
+                                                  Operation *bias_add_op) {
+  // Fused op name dependes on original contraction op name.
+  std::string fused_op_name;
+  if (helper.getDialect()->IsConv2D(contraction_op)) {
+    fused_op_name = "tfg._FusedConv2D";
+  } else if (helper.getDialect()->IsMatMul(contraction_op)) {
+    fused_op_name = "tfg._FusedMatMul";
+  } else if (helper.getDialect()->IsDepthwiseConv2dNative(contraction_op)) {
+    fused_op_name = "tfg._FusedDepthwiseConv2dNative";
+  } else if (helper.getDialect()->IsConv3D(contraction_op)) {
+    fused_op_name = "tfg._FusedConv3D";
+  } else {
+    return rewriter.notifyMatchFailure(contraction_op,
+                                       "Unsupported contraction op.");
+  }
+
+  SmallVector<Value, 4> operands;
```

Comment:
Just a nit: in general you don't need to provide the size of a SmallVector.

Reply:
Done.